pandas/core/dtypes/cast.py

```diff
@@ -1307,9 +1309,16 @@ def maybe_infer_to_datetimelike(
        leave inferred dtype 'date' alone
 
     """
-    # TODO: why not timedelta?
     if isinstance(
-        value, (ABCDatetimeIndex, ABCPeriodIndex, ABCDatetimeArray, ABCPeriodArray)
+        value,
+        (
+            ABCDatetimeArray,
+            ABCDatetimeIndex,
+            ABCPeriodArray,
+            ABCPeriodIndex,
+            ABCTimedeltaArray,
+            ABCTimedeltaIndex,
+        ),
```

Comment:
If this check is updated, then the check on L1325 should also be updated (doubly-so because it is out of date, should be ABCDatetimeArray instead of ABCDatetimeIndex)

Reply:
Ok! Will circle back and do this if we're keeping this form of the check

Reply:
Updated this

Reply:
looks like the update didnt get pushed?

Reply:
I switched `DatetimeIndex` -> `DatetimeArray` on the line below.

The changes to the first check that we discussed give perf degradations, e.g. https://github.com/pandas-dev/pandas/pull/37236#issuecomment-715406560